pkg/nvmf/fabrics.go

```diff
@@ -115,24 +168,55 @@ func disconnectSubsys(nqn, ctrl string) (res bool) {
 	return true
 }
 
-func disconnectByNqn(nqn string) int {
+func disconnectByNqn(nqn, hostnqn string) int {
 	ret := 0
 	if len(nqn) > NVMF_NQN_SIZE {
 		klog.Errorf("Disconnect: nqn %s is too long ", nqn)
 		return -EINVAL
 	}
 
+	// delete hostnqn file
+	hostnqnPath := strings.Join([]string{RUN_NVMF, nqn, b64.StdEncoding.EncodeToString([]byte(hostnqn))}, "/")
```

Comment:
Okay, I will use path.Join() instead of strings.join(), but still we need base64 encoding. Because a hostnqn which is NodeUnpublishVolumeRequest's targetPath contains "/" characters.